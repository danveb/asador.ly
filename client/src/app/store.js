import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../redux/auth/authSlice";
import pinReducer from "../redux/pins/pinSlice";

export const store = configureStore({
  reducer: {
    auth: authReducer,
    pins: pinReducer,
  },
}); 