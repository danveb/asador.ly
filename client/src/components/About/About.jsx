import { Link } from "react-router-dom"; 
import { Navbar, Menu, Footer } from "../index"; 
import brasa from "../../assets/about/brasa.png"; 
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import "./About.scss"; 

const About = ({ menuOpen, setMenuOpen }) => {
    return (
        <div>
            <Navbar menuOpen={menuOpen} setMenuOpen={setMenuOpen} />
            <Menu menuOpen={menuOpen} setMenuOpen={setMenuOpen} />
            <div className="about">
                <div className="about-top">
                    <div className="about-title">
                        <h3>Explore...</h3>
                        <h1>Buenos Aires</h1>
                        <Link to="/">Check Our Map</Link>
                    </div>
                    <div className="about-img">
                        <img src={brasa} alt="charcoal" />
                    </div>
                    <div className="arrow">
                        <KeyboardArrowDownIcon />
                    </div>
                </div>
                <div className="about-bottom">
                    <div className="about-subtitle">
                        <h2>Asador</h2>
                        <p>It's all about sharing the joy of <span>asado</span> (barbeque) that  makes Argentina a prime destination for dining at <span>parrillas</span> (steakhouses). From traditional eateries to fine-dining establishments, there's always a <span>parrilla</span> to choose from for a fix of authentic Argentine <span>asado</span>. </p>
                        <p>Here at <span>Asador</span> we have curated a list we recommend of the best steakhouses in Buenos Aires (we may have missed a few, on purpose, just so you can share your experience with us too) for every budget. Carnivores rejoice!</p> 
                    </div>
                </div>
            </div>
            <Footer />
        </div>
    )
}

export default About