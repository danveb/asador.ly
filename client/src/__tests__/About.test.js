import { render } from "@testing-library/react"; 
import '@testing-library/jest-dom/extend-expect';
import { store } from "../app/store"; 
import { Provider } from "react-redux";
import About from "../components/About/About"; 
import { BrowserRouter } from "react-router-dom";

describe("About component", () => {
    // smoke test
    test("renders About component", () => {
        render(
            <Provider store={store}>
                <BrowserRouter>
                    <About />
                </BrowserRouter>
            </Provider>
        );
    });

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <About />
                </BrowserRouter>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText
    test("displays correct h1 title", () => {
        const { getByText } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <About />
                </BrowserRouter>
            </Provider>
        );
        const title = getByText("Buenos Aires"); 
        expect(title).not.toBe(<h3>Buenos Aires</h3>); 
    });

    // getByAltText
    test("img tag should have alt attribute", () => {
        const { getByAltText } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <About />
                </BrowserRouter>
            </Provider>
        );
        const altText = getByAltText("charcoal"); 
        expect(altText).toBeInTheDocument(); 
    }); 

    // getByAltText
    test("throws error with incorrect alt attribute", () => {
        const { getByAltText } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <About />
                </BrowserRouter>
            </Provider>
        );
        const altText = getByAltText("charcoal"); 
        expect(altText).not.toBe("brasa"); 
    }); 
});