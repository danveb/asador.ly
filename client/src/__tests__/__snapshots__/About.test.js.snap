// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="navbar undefined"
    >
      <div
        class="navbar-left"
      >
        <a
          href="/"
        >
          Asador
        </a>
      </div>
      <div
        class="navbar-right"
      >
        <ul
          class="navbar-links"
        >
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </a>
          </li>
        </ul>
        <div
          class="hamburger"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
    <div
      class="menu undefined"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/login"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AccountCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="about"
    >
      <div
        class="about-top"
      >
        <div
          class="about-title"
        >
          <h3>
            Explore...
          </h3>
          <h1>
            Buenos Aires
          </h1>
          <a
            href="/"
          >
            Check Our Map
          </a>
        </div>
        <div
          class="about-img"
        >
          <img
            alt="charcoal"
            src="brasa.png"
          />
        </div>
        <div
          class="arrow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </div>
      </div>
      <div
        class="about-bottom"
      >
        <div
          class="about-subtitle"
        >
          <h2>
            Asador
          </h2>
          <p>
            It's all about sharing the joy of 
            <span>
              asado
            </span>
             (barbeque) that  makes Argentina a prime destination for dining at 
            <span>
              parrillas
            </span>
             (steakhouses). From traditional eateries to fine-dining establishments, there's always a 
            <span>
              parrilla
            </span>
             to choose from for a fix of authentic Argentine 
            <span>
              asado
            </span>
            . 
          </p>
          <p>
            Here at 
            <span>
              Asador
            </span>
             we have curated a list we recommend of the best steakhouses in Buenos Aires (we may have missed a few, on purpose, just so you can share your experience with us too) for every budget. Carnivores rejoice!
          </p>
        </div>
      </div>
    </div>
    <div
      class="footer-container"
    >
      <div
        class="footer-left"
      >
        <h3>
          © All rights reserved – Danny Bae
        </h3>
      </div>
      <div
        class="footer-right"
      >
        <a
          href="https://www.github.com/danveb"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="GitHubIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
